domain/schema/model.go

```diff
@@ -124,6 +124,24 @@ func ModelDDL() *schema.Schema {
 	patches = append(patches,
 		triggersForImmutableTable("model", "", "model table is immutable"),
 
+		// The charm is unmodifiable.
+		// There is a lot of assumptions in the code that the charm is immutable
+		// from modification. This is a safety net to ensure that the charm is
+		// not modified.
+		triggersForUnmodifiableTable("charm_action", "charm action table is immutable"),
```

Comment:
Why not use the real table name in the error message?

I would also be a little clearer, stating that the _rows_ are not updatable.

Reply:
Also `unmodifiable` vs `immutable`